specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/preview/2023-07-01-preview/virtualMachineInstances.json

```diff
@@ -0,0 +1,1545 @@
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    }
+  },
+  "definitions": {
+    "VirtualMachineInstanceProperties": {
+      "type": "object",
+      "x-ms-client-flatten": true,
+      "properties": {
+        "hardwareProfile": {
+          "type": "object",
+          "properties": {
+            "vmSize": {
+              "type": "string",
+              "enum": [
+                "Default",
+                "Standard_A2_v2",
+                "Standard_A4_v2",
+                "Standard_D2s_v3",
+                "Standard_D4s_v3",
+                "Standard_D8s_v3",
+                "Standard_D16s_v3",
+                "Standard_D32s_v3",
+                "Standard_DS2_v2",
+                "Standard_DS3_v2",
+                "Standard_DS4_v2",
+                "Standard_DS5_v2",
+                "Standard_DS13_v2",
+                "Standard_K8S_v1",
+                "Standard_K8S2_v1",
+                "Standard_K8S3_v1",
+                "Standard_K8S4_v1",
+                "Standard_NK6",
+                "Standard_NK12",
+                "Standard_NV6",
+                "Standard_NV12",
+                "Standard_K8S5_v1",
+                "Custom"
+              ],
+              "default": "Default",
+              "x-ms-enum": {
+                "name": "VmSizeEnum",
+                "modelAsString": true
+              }
+            },
+            "processors": {
+              "type": "integer",
+              "format": "int32",
+              "description": "number of processors for the virtual machine instance"
+            },
+            "memoryMB": {
+              "type": "integer",
+              "format": "int64",
+              "description": "RAM in MB for the virtual machine instance"
+            },
+            "dynamicMemoryConfig": {
+              "type": "object",
+              "properties": {
+                "maximumMemoryMB": {
+                  "type": "integer",
+                  "format": "int64"
+                },
+                "minimumMemoryMB": {
+                  "type": "integer",
+                  "format": "int64"
+                },
+                "targetMemoryBuffer": {
+                  "type": "integer",
+                  "description": "Defines the amount of extra memory that should be reserved for a virtual machine instance at runtime, as a percentage of the total memory that the virtual machine instance is thought to need. This only applies to virtual systems with dynamic memory enabled. This property can be in the range of 5 to 2000.",
+                  "format": "int32"
+                }
+              }
```

Comment:
Any min/max/default for these values?

Reply:
dynamicMemoryConfig is only relevant when VM does not use a static memory. We could add a min max but giving it a default value will make it always return a targetMemoryBuffer which is not relevant in static memory.

Reply:
We are trying to stay in sync with some partner RPs (like [SCVMM ](https://github.com/Azure/azure-rest-api-specs-pr/blob/bb15ae0084afda87b7d50e3e5a0e4ac66d0e5f9d/specification/scvmm/resource-manager/Microsoft.ScVmm/preview/2023-04-01-preview/scvmm.json#L4939C16-L4939C16)) who are implementing the same contract for equivalent child virtualMachineInstance resources. They do not include a min, max, or default value. So I would prefer not to be different than they are.

Reply:
Resolving this since @zizw123 said to just apply what was appropriate:

<img width="743" alt="image" src="https://github.com/Azure/azure-rest-api-specs/assets/84938589/c5bff601-163e-455d-996b-a4dd3e5cf018">